DataFormats/HcalRecHit/interface/HFPreRecHit.h

```diff
@@ -28,6 +30,29 @@ class HFPreRecHit
     // index is out of range.
     const HFQIE10Info* getHFQIE10Info(unsigned index) const;
 
+    // Quantities simply added from both anodes
+    float charge() const;
+    float energy() const;
```

Comment:
This sum is needed in more than one place, so it is best to have
it in this class -- just the basic OO principles.

On 08/04/2016 04:57 PM, Slava Krutelyov wrote:
>
> In DataFormats/HcalRecHit/interface/HFPreRecHit.h <https://github.com/cms-sw/cmssw/pull/15324#discussion_r73610270>:
>
> > @@ -28,6 +30,29 @@ class HFPreRecHit
> >      // index is out of range.
> >      const HFQIE10Info* getHFQIE10Info(unsigned index) const;
> >
> > +    // Quantities simply added from both anodes
> > +    float charge() const;
> > +    float energy() const;
>
> will this remain "simply added" ?
> Initially, PreRecHit looked like a slightly wrapped access to 1 or 2 HFQIE10Info.
> Then, all proper combinations into a single observable would be in HFRecHit.
>
> Maybe make it chargeTotal and energyTotal?
>
> .. it's minimal enough for now to stay as is, but it seems to me like new functionality is creeping in
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324/files/f2734c7b29b226480f1efc887de88c8ae73f121f#r73610270>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-Q4SOATYQRW4Uil7WMzuLPxs8fRmks5qcmBGgaJpZM4JXio9>.
>